src/lib/ui.js

```diff
@@ -0,0 +1,89 @@
+import config from '../config'
+import urltemplate from 'url-template'
+import { awaitMessage } from '../utils/async'
+
+const uiMethods = async({ method, href } = {}) => {
+  switch (method) {
+    case 'share':
+      if (typeof href === 'string') {
+        // we didn't do any validation but left it for sharing page to validate it.
+        const dialog = openWindow(encodeURIComponent(href))
+        return awaitMessage(dialog)
+      } else {
```

Comment:
It is possible to remove this else statement and just ```throw Error(`href: ${href} is not valid.`)``` since there is a `return` in the if statement and no other code will run apart from the `throw`